src/odemis/model/_vattributes.py

```diff
@@ -982,6 +983,30 @@ def _check(self, value):
         Enumerated._check(self, value)
         VigilantAttribute._check(self, value)
 
+    def clip(self, val):
+        """ Clip the given value to fit within the choices.
+
+        If the range contains vectors of length n, each element of val will be
+        clipped separately by position for positions 0..n-1
+        return (same type as val): value clipped
+        """
+
+        if isinstance(val, tuple):
+            if val not in self.choices:
+                dist = distance.euclidean(val, max(self.choices))
```

Comment:
You're just looking for the closest number, using the euclidean.distance. In Python, that can be written in a shorter way, as `util.find_closest()`. Roughly like:
`min(self.choices, key=lambda x: distance.euclidean(x - val))`
If you want to skip the choices which are not compatible with the value (and so distance.euclidean raises an error), you can define a small method which calls distance.euclidean and returns infinite if it's raising an Error.

(You might want to also check where all the choices fail to compare to the proposed value, in which case, it'd return the current value, instead of returning a "random" choice).